### ISSUE TITLE
Change Function Convert-SmbiosUuid to do correct ByteSwapping

### ISSUE THREAD

**Issue by chris8205** [csc/Hanlon#419]
```ruby
Function Convert-SmbiosUuid {

Param (
	[Parameter(Mandatory=$True)]
	[String]
	$rawUUID
)


# Create an array of each half (hyphen delimiter)
$octets = $rawUUID.Split("-")

# Create an array of each two-charactere byte (space delimiter)
$bytes = $octets[0].Split(" ") + $octets[1].Split(" ")

# Build the final string, piecing together byte by byte
$prettyUUID = $bytes[3] + $bytes[2] + $bytes[1] + $bytes[0] + "-" + $bytes[5] + $bytes[4] + "-" + $bytes[7] + $bytes[6] + "-" + $bytes[8] + $bytes[9] + "-" + $bytes[10] + $bytes[11] + $bytes[12] + $bytes[13] + $bytes[14] + $bytes[15]

Return $prettyUUID

}
```

**Comment by chris8205** [comment]
Or implement something like this;
maybe it isn't the proper way but...

```ruby
Function Get-SmbiosUuid {

if ((Get-WmiObject win32_computersystemproduct).Vendor -like "Innotek*")
{
# on Oracle VM it seems, that the UUID is not in the right byte order
# to test with other versions of Oracle VM / other vendors
# Build the final string, piecing together Byte by Byte 
$bytes=(Get-WmiObject win32_computersystemproduct).uuid.Split("-") -split '(..)' | ? { $_ };
$prettyUUID = $bytes[3] + $bytes[2] + $bytes[1] + $bytes[0] + "-" + $bytes[5] + $bytes[4] + "-" + $bytes[7] + $bytes[6] + "-" + $bytes[8] + $bytes[9] + "-" + $bytes[10] + $bytes[11] + $bytes[12] + $bytes[13] + $bytes[14] + $bytes[15];
}
else
{
# for all other systems take the original uuid from WMI-Object
$prettyUUID = (Get-WmiObject win32_computersystemproduct).uuid;
}

return $prettyUUID;

}
```

**Comment by jcpowermac** [comment]
@chris8205 I am reviewing the code and results from multiple hypervisors.  I will try to get an update soon.

**Comment by jcpowermac** [comment]
@chris8205 check out 
https://github.com/jcpowermac/Hanlon/blob/windows_issue_419/scripts/winpe/hanlon-discover.ps1
and see if that works.

**Comment by chris8205** [comment]
No...

its returning ffffffxx in the first octet, the rest is fine but lowercase...

Am 07.01.2016 um 03:54 schrieb Joseph Callen:

**Comment by jcpowermac** [comment]
@chris8205 try again.  I got caught in an signed vs unsigned integer issue.

**Comment by tjmcs** [comment]
Any feedback on PR #420, @chris8205? Waiting to hear back from you (or anyone else who's testing it) before merging...

**Comment by chris8205** [comment]
Will Test it later, actually to busy...
with one of my suggestions its Running fine, will try your conversion tomorrow...
Need to rebuild the .wim ......

**Comment by tjmcs** [comment]
Understand completely, @chris8205; @jcpowermac will be testing as well but wanted to get a chance for you to provide your feedback (since it's an issue you uncovered) before merging anything; just let me know when you've had a chance to test it.

**Comment by tjmcs** [comment]
@chris8205; this issue was closed by the merging of PR #420, which was submitted (and tested) by @jcpowermac late last week. If you are still having problems deploying Windows to your VirtualBox instances, feel free to reopen it, but if this remains closed I'll assume that the merging of that PR resolved the issue you described here. Thanks again for uncovering and reporting this issue :)
